plugins/woocommerce/src/Admin/API/Leaderboards.php

```diff
@@ -53,6 +53,22 @@ public function register_routes() {
 			)
 		);
 
+		foreach ( [ 'customers', 'coupons', 'categories', 'products' ] as $endpoint ) {
+			register_rest_route(
+				$this->namespace,
+				'/' . $this->rest_base . '/' . $endpoint,
+				array(
+					array(
+						'methods'             => \WP_REST_Server::READABLE,
+						'callback'            => array( $this, 'get_items' ),
+						'permission_callback' => array( $this, 'get_items_permissions_check' ),
+						'args'                => $this->get_collection_params(),
+					),
+					'schema' => array( $this, 'get_public_item_schema' ),
+				)
+			);
+		}
```

Comment:
Instead of creating four separate endpoints, one alternative would be to use a URI argument instead. Something like this:

```php
		register_rest_route(
				$this->namespace,
				'/' . $this->rest_base . '/(?P<leaderboard>[\w-]+)',
				array(
					'args' => array(
						'leaderboard' => array(
							'type' => 'string',
							'enum' => array( 'customers', 'coupons', 'categories', 'products' ),
						),
					)
					array(
						'methods'             => \WP_REST_Server::READABLE,
						'callback'            => array( $this, 'get_items' ),
						'permission_callback' => array( $this, 'get_items_permissions_check' ),
						'args'                => $this->get_collection_params(),
					),
					'schema' => array( $this, 'get_public_item_schema' ),
				)
			);
```

I'm not sure this has any huge advantages over the other approach, but it does simplify things a bit by having less total endpoints, and gives you built-in validation and error handling for the endpoint URI, which seems a little less fragile than traversing to the end of an array to determine which leaderboard to show.